inference-engine/src/mkldnn_plugin/nodes/mkldnn_detection_output_node.cpp

```diff
@@ -125,480 +152,753 @@ void MKLDNNDetectionOutputNode::initSupportedPrimitiveDescriptors() {
-                decode_bbox_height   = std::exp(variance_data[p*4 + 3] * loc_ymax) * prior_height;
+                decodeBboxCenterX = varianceData[p*4 + 0] * locXMin * priorWidth + priorCenterX;
+                decodeBboxCenterY = varianceData[p*4 + 1] * locYMin * priorHeight + priorCenterY;
+                decodeBboxWidth    = std::exp(varianceData[p*4 + 2] * locXMax) * priorWidth;
+                decodeBboxHeight   = std::exp(varianceData[p*4 + 3] * locYMax) * priorHeight;
             }
 
-            new_xmin = decode_bbox_center_x - decode_bbox_width  / 2.0f;
-            new_ymin = decode_bbox_center_y - decode_bbox_height / 2.0f;
-            new_xmax = decode_bbox_center_x + decode_bbox_width  / 2.0f;
-            new_ymax = decode_bbox_center_y + decode_bbox_height / 2.0f;
+            newXMin = decodeBboxCenterX - decodeBboxWidth  / 2.0f;
+            newYMin = decodeBboxCenterY - decodeBboxHeight / 2.0f;
+            newXMax = decodeBboxCenterX + decodeBboxWidth  / 2.0f;
+            newYMax = decodeBboxCenterY + decodeBboxHeight / 2.0f;
         }
 
-        if (_clip_before_nms) {
-            new_xmin = (std::max)(0.0f, (std::min)(1.0f, new_xmin));
-            new_ymin = (std::max)(0.0f, (std::min)(1.0f, new_ymin));
-            new_xmax = (std::max)(0.0f, (std::min)(1.0f, new_xmax));
-            new_ymax = (std::max)(0.0f, (std::min)(1.0f, new_ymax));
+        if (clipBeforeNMS) {
+            newXMin = (std::max)(0.0f, (std::min)(1.0f, newXMin));
+            newYMin = (std::max)(0.0f, (std::min)(1.0f, newYMin));
+            newXMax = (std::max)(0.0f, (std::min)(1.0f, newXMax));
+            newYMax = (std::max)(0.0f, (std::min)(1.0f, newYMax));
         }
 
-        decoded_bboxes[p*4 + 0] = new_xmin;
-        decoded_bboxes[p*4 + 1] = new_ymin;
-        decoded_bboxes[p*4 + 2] = new_xmax;
-        decoded_bboxes[p*4 + 3] = new_ymax;
+        decodedBboxes[p*4 + 0] = newXMin;
+        decodedBboxes[p*4 + 1] = newYMin;
+        decodedBboxes[p*4 + 2] = newXMax;
+        decodedBboxes[p*4 + 3] = newYMax;
 
-        decoded_bbox_sizes[p] = (new_xmax - new_xmin) * (new_ymax - new_ymin);
+        decodedBboxSizes[p] = (newXMax - newXMin) * (newYMax - newYMin);
     });
 }
 
-void MKLDNNDetectionOutputNode::nms_cf(const float* conf_data,
-                                 const float* bboxes,
-                                 const float* sizes,
-                                 int* buffer,
-                                 int* indices,
-                                 int& detections,
-                                 int num_priors_actual) {
-    int count = 0;
-    for (int i = 0; i < num_priors_actual; ++i) {
-        if (conf_data[i] > _confidence_threshold) {
-            indices[count] = i;
-            count++;
-        }
+inline void MKLDNNDetectionOutputNode::topk(const int *indicesIn, int *indicesOut, const float *conf, int n, int k) {
```

Comment:
Don't really think we can overtake partial_sort_copy but still does it make sense to try optimized TopK code @xuchen-intel has just implemented?

Reply:
I will discuss if current topk can be apply to this case with xuchen.  If supported, I will have a compare the perf. Anyway, this will be done in a separate PR after topk is finalized. Thank you!